### PR TITLE
feat: add core unified prefix for Elasticsearch/OpenSearch

### PR DESCRIPTION
**Pull request by aabouzaid**
### Which problem does the PR fix?

Task: https://github.com/camunda/camunda-platform-helm/issues/2610

### What's in this PR?

Added core unified prefix for Elasticsearch/OpenSearch.

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/docs/contributing.md).

**Before opening the PR:**

- [ ] In the repo's root dir, run `make go.update-golden-only`.
- [ ] There is no other open [pull request](../pulls) for the same update/change.
- [ ] Tests for charts are added (if needed).
- [ ] In-repo [documentation](../blob/main/docs/contributing.md#documentation) are updated (if needed).

**After opening the PR:**

- [ ] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [ ] Did all checks/tests pass in the PR?

### CODE REVIEW

File: charts/camunda-platform-alpha/templates/core/configmap.yaml [hunk 1]
```diff
@@ -263,7 +271,9 @@ data:
           # Transport port
           port: {{ .Values.global.elasticsearch.url.port }}
           # Index prefix, configured in Zeebe Elasticsearch exporter
-          prefix: {{ .Values.global.elasticsearch.prefix }}
+          {{- if .Values.core.index.prefix }}
+          prefix: {{ .Values.core.index.prefix | quote }}
+          {{- end }}
```
Comment (ChrisKujawa) [inline charts/camunda-platform-alpha/templates/core/configmap.yaml:276]: ❌ I think here we wanted to keep the other prefix right?

File: charts/camunda-platform-alpha/templates/optimize/configmap.yaml [hunk 2]
```diff
@@ -23,6 +23,11 @@ data:
 
     {{- if .Values.global.elasticsearch.enabled }}
     es:
+      settings:
+        {{- if .Values.core.index.prefix }}
+        index:
+          prefix: {{ .Values.core.index.prefix | quote }}
+        {{- end }}
```
Comment (ChrisKujawa) [inline charts/camunda-platform-alpha/templates/optimize/configmap.yaml:30]: ❌ If we use for the ES exporter different prefix, then likely we need to use here the same for optimize

File: charts/camunda-platform-alpha/templates/core/configmap.yaml (hunk 1 above)
Comment (aabouzaid) [reply charts/camunda-platform-alpha/templates/core/configmap.yaml:276]: Fixed in: https://github.com/camunda/camunda-platform-helm/commit/ae9c82512781c1c307eb20a96589d0b7575aa3e0

File: charts/camunda-platform-alpha/templates/optimize/configmap.yaml (hunk 2 above)
Comment (aabouzaid) [reply charts/camunda-platform-alpha/templates/optimize/configmap.yaml:30]: Fixed in: https://github.com/camunda/camunda-platform-helm/commit/ae9c82512781c1c307eb20a96589d0b7575aa3e0

**Pull request merged**